octavia_f5/utils/cert_manager.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2019 SAP SE
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import hashlib
+
+from oslo_config import cfg
+from oslo_context import context as oslo_context
+from stevedore import driver as stevedore_driver
+
+from octavia.common.tls_utils import cert_parser
+from octavia_f5.common import constants
+from octavia_f5.restclient.as3objects import certificate as m_cert
+
+CONF = cfg.CONF
+
+
+class CertManager(object):
+    def __init__(self):
+        self.cert_manager = stevedore_driver.DriverManager(
+            namespace='octavia.cert_manager',
+            name=CONF.certificates.cert_manager,
+            invoke_on_load=True,
+        ).driver
+
+    def get_certificates(self, obj, context=None):
+        certificates = []
+        cert_dict = cert_parser.load_certificates_data(self.cert_manager, obj, context)
+        if obj.tls_certificate_id:
+            cert_dict['container_id'] = obj.tls_certificate_id.split('/')[-1]
+
+        # Note, the first cert is the TLS default cert
```

Comment:
afaik not a default tls certificate is always required regardless of SNI certs or not

Reply:
In that case, writing an if-clause would cause the program to fail silently when no TLS certificate is present. I think it would be better to remove the if-clause, what do you think?